### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by ksigei**
### Create models
-  Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
-  Set up associations between models.

### Use models to insert data
- Use rails console to test data in the project.
- Create user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(user: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create posts comments for one of the posts by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.
-Use `rails c` to confirm if my methods are working.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,8 +1,21 @@
 Rails.application.routes.draw do
   root "posts#index"
-  get 'comments/comments'
-  get 'likes/likes'
-  get 'users/users'
+  resources :posts do 
+    # resources :comments
```
Comment (rustam-tolipov) [inline config/routes.rb]: - [ ] Kindly remove unused commented code from your project.

File: config/routes.rb [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+  resources :users do
+    resources :posts do
+      resources :comments
+    end
+  end
+  
+  get '/about', to: 'about#about'
+  # get 'comments/comments'
+  # get 'likes/likes'
+  # get 'users/users'
```
Comment (rustam-tolipov) [inline config/routes.rb]: - [ ] Kindly remove unused commented codes.

File: app/models/post.rb [hunk 3]
```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  belongs_to :user
+
+  def update_posts_counter
```
Comment (rustam-tolipov) [inline app/models/post.rb:13]: - [ ] It's good practice if you put `update_posts_counter` after the private method. 
[More info...](https://guides.rubyonrails.org/active_record_callbacks.html#:~:text=before_validation%20%3Aensure_login_has_a_value-,private,-def%20ensure_login_has_a_value%0A%20%20%20%20%20%20if)

File: app/models/like.rb [hunk 4]
```diff
@@ -1,2 +1,10 @@
 class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_likes_counter
```
Comment (rustam-tolipov) [inline app/models/like.rb:8]: - [ ] It's good practice if you put `update_likes_counter` after the private method. And call it `after_create`.
[More info...](https://guides.rubyonrails.org/active_record_callbacks.html#:~:text=before_validation%20%3Aensure_login_has_a_value-,private,-def%20ensure_login_has_a_value%0A%20%20%20%20%20%20if)

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @ksigei, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @ksigei  😊 😃 

## [OPTIONAL]

- [ ] Please use the README template provided by Microverse and Kindly also add the following in the README file:
- Project Title
- Description
- Tech Stack
- Set of steps to run your project on a local machine
- Social Media Accounts.

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**